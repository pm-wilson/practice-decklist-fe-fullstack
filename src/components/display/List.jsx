import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const List = ({ 
  decks,
  newDeckName,
  newDeckColors,
  newDeckFormat,
  handleChange,
  handleSubmit,
}) => {

  //needs to update state when form is changed
  //needs to update decklist when form is submitted
  //needs to rerender page
  return (
    <div>   
      <form onSubmit={handleSubmit}>
        <h2>New Deck</h2>   
        <label>Deck Name
          <input name="deck" onChange={handleChange} value={newDeckName}></input>
        </label>
        <label>Colors
          <input name="colors" onChange={handleChange} value={newDeckColors}></input>
        </label>
        <label>Format
          <input name="format" onChange={handleChange} value={newDeckFormat}></input>
        </label>
        <button>Add Deck</button>
      </form>
      <div>
        <h2>Deck List</h2>
        {decks.map(deck => {
          return (
            <div key={deck.id}>
              <Link to={`/details/${deck.id}`}>{deck.deck}</Link>
            </div>
          );
        })}
      </div>
    </div>
  );
};

List.propTypes = {
  decks: PropTypes.array,
  newDeckName: PropTypes.string,
  newDeckColors: PropTypes.string,
  newDeckFormat: PropTypes.string,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
};

export default List;
