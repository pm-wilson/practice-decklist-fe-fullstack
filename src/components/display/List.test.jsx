import React from 'react';
import { render } from '@testing-library/react';
import List from './List';
import { MemoryRouter } from 'react-router';

describe('List component', () => {
  const decks = [{ id: 1, name: 'Balthazor' }, { id: 2, name: 'Baru' }];

  it('renders List', () => {
    const { asFragment } = render(<MemoryRouter><List decks={decks} /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
