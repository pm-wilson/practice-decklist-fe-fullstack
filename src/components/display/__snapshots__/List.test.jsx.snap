// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h2>
        New Deck
      </h2>
      <label>
        Deck Name
        <input
          name="deck"
          value=""
        />
      </label>
      <label>
        Colors
        <input
          name="colors"
          value=""
        />
      </label>
      <label>
        Format
        <input
          name="format"
          value=""
        />
      </label>
      <button>
        Add Deck
      </button>
    </form>
    <div>
      <h2>
        Deck List
      </h2>
      <div>
        <a
          href="/details/1"
        />
      </div>
      <div>
        <a
          href="/details/2"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
